controllers/state_of_the_world.go

```diff
@@ -86,9 +136,24 @@ func NewPolicyMachineryController(manager ctrlruntime.Manager, client *dynamic.D
 		logger.Info("envoygateway is not installed, skipping related watches and reconcilers", "err", err)
 	} else {
 		controllerOpts = append(controllerOpts,
-			controller.WithRunnable("envoypatchpolicy watcher", controller.Watch(&egv1alpha1.EnvoyPatchPolicy{}, envoygateway.EnvoyPatchPoliciesResource, metav1.NamespaceAll)),
-			controller.WithRunnable("envoyextensionpolicy watcher", controller.Watch(&egv1alpha1.EnvoyExtensionPolicy{}, envoygateway.EnvoyExtensionPoliciesResource, metav1.NamespaceAll)),
-			controller.WithRunnable("envoysecuritypolicy watcher", controller.Watch(&egv1alpha1.SecurityPolicy{}, envoygateway.SecurityPoliciesResource, metav1.NamespaceAll)),
+			controller.WithRunnable("envoypatchpolicy watcher", controller.Watch(
+				&egv1alpha1.EnvoyPatchPolicy{},
+				envoygateway.EnvoyPatchPoliciesResource,
+				metav1.NamespaceAll,
+				controller.WithPredicates(&ctrlruntimepredicate.TypedGenerationChangedPredicate[*egv1alpha1.EnvoyPatchPolicy]{})),
+			),
+			controller.WithRunnable("envoyextensionpolicy watcher", controller.Watch(
+				&egv1alpha1.EnvoyExtensionPolicy{},
+				envoygateway.EnvoyExtensionPoliciesResource,
+				metav1.NamespaceAll,
+				controller.WithPredicates(&ctrlruntimepredicate.TypedGenerationChangedPredicate[*egv1alpha1.EnvoyExtensionPolicy]{})),
+			),
+			controller.WithRunnable("envoysecuritypolicy watcher", controller.Watch(
+				&egv1alpha1.SecurityPolicy{},
+				envoygateway.SecurityPoliciesResource,
+				metav1.NamespaceAll,
+				controller.WithPredicates(&ctrlruntimepredicate.TypedGenerationChangedPredicate[*egv1alpha1.SecurityPolicy]{})),
+			),
```

Comment:
Same https://github.com/Kuadrant/kuadrant-operator/pull/885/files#r1787506477

Reply:
Will remove this predicate, and also for this isio CRDs since I assume we want status changes for those also